src/feature_reader.rs

```diff
@@ -0,0 +1,271 @@
+use crate::de::deserialize_feature_collection;
+use crate::{Feature, Result};
+
+use serde::de::DeserializeOwned;
+
+use std::io::Read;
+
+pub struct FeatureReader<R> {
```

Comment:
Could we get a short doc string here so it's visible in the docs?
<img width="910" alt="Screenshot 2022-08-25 at 14 05 27" src="https://user-images.githubusercontent.com/131862/186672679-c67a4626-d980-4b88-87d1-5adb9e32615d.png">